### ISSUE TITLE
centering image in page

### ISSUE THREAD

**Issue by vanboom** [mogest/prawn-svg#52]
Would it be possible to add an align option to allow :left | :right | :center justification of the image in the page? 

Thank you thank you thank you!  Great gem!

**Comment by vanboom** [comment]
I just figured out... we can align by fixing the width then calculating the X position.   So the :align option would only be necessary if you want to support auto-sized images.

**Comment by mojavelinux** [comment]
I still think it would be worthwhile to support centering within the bounding box to align with the `image` method in Prawn.

Here's the logic I currently use:

```ruby
left = case position
when :left 
  0
when :right
  bounds.width - width 
when :center
  ((bounds.width - width) / 2.0).floor
end

svg ::IO.read(image_path), at: [left, cursor], width: width
```

Of course, it depends on knowing the width in advance.

**Comment by mogest** [comment]
In most use cases of prawn-svg I've seen, the implementor has wanted to place it exactly at a certain place on the page.  I'm open to the idea of adding an `align` option that does exactly what your code does, though.  I'd leave it defaulting to `:left`.  Thoughts?

**Comment by mojavelinux** [comment]
@mogest the default option should definitely be left.

Oddly enough, Prawn use the attribute `:position` for images instead of align. Perhaps Prawn SVG can “align” with Prawn core (or perhaps allow either key name). Keeping consistent with core allows the same configuration parameters to be used for either raster or SVG formats.

As for the requirements of implementers, I can say with certainty that once we start talking about laying out arbitrary documents, horizontal centering is a must have. So this is an essential feature for Prawn SVG if it's going to be used in that context (unless the implementer takes on the responsibility, as I have done thus far in Asciidoctor PDF).

**Comment by mojavelinux** [comment]
While we're talking about alignment, it might be nice if `:at` isn't required. It should default to the calculated position, with `:left` as the default, like Prawn's built-in image function. See https://github.com/prawnpdf/prawn/blob/master/lib/prawn/images.rb#L110-L115.

**Comment by mojavelinux** [comment]
Can we reopen the issue since we're reconsidering this feature now?

**Comment by mojavelinux** [comment]
Thanks! If you need assistance, just let me know. Cheers!

**Comment by mogest** [comment]
If you wouldn't mind having a look at the commit, I'd appreciate it.

Specifically, I've decided not to floor the centered position; I think Prawn and PDFs deal fine with floating point numbers.  Anything I should be worried about that you know of?

**Comment by mojavelinux** [comment]
Looks great to me!

floor isn't necessary. I can't really remember why I added that except maybe I was testing something and left it in there :)

Any chance we could get vposition next? I'll file an issue if you +1 the idea.

**Comment by mogest** [comment]
Done, built and released 0.19.0.
